src/main/java/org/victorrobotics/frc/dtlib/controller/DTController.java

```diff
@@ -0,0 +1,200 @@
+package org.victorrobotics.frc.dtlib.controller;
+
+import org.victorrobotics.frc.dtlib.exception.DTIllegalArgumentException;
+
+import java.util.Arrays;
+
+import edu.wpi.first.hal.DriverStationJNI;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj.DriverStation;
+
+public abstract class DTController {
+    private static final int MAX_CONTROLLER_COUNT = 6;
+
+    private static final DTController[] INSTANCES = new DTController[MAX_CONTROLLER_COUNT];
+
+    private final int port;
+
+    private double[] axesCurrent;
+    private double[] axesPrevious;
```

Comment:
Caching is a good idea, however, WPILib already caches the values [here](https://github.com/wpilibsuite/allwpilib/blob/a791470de77b3e02e67258f03c57e2a4edf1dae0/wpilibj/src/main/java/edu/wpi/first/wpilibj/DriverStation.java#L1358)

Reply:
Also, it doesn’t remember previous values for axes or POV, which could prove useful